pkg/parse/state.go

```diff
@@ -104,42 +118,19 @@ func (s *reconcilerState) reset() {
 	s.resetCache()
 	s.lastApplied = ""
 	s.cache.needToRetry = true
-	s.cache.nextRetryTime = time.Now().Add(time.Second)
+	s.backoff = defaultBackoff()
 }
 
 // invalidate logs the errors, clears the state tracking information.
 // invalidate does not clean up the `s.cache`.
 func (s *reconcilerState) invalidate(errs status.MultiError) {
 	klog.Errorf("Invalidating reconciler checkpoint: %v", status.FormatSingleLine(errs))
-	oldErrs := s.cache.errs
 	s.cache.errs = errs
 	// Invalidate state on error since this could be the result of switching
 	// branches or some other operation where inverting the operation would
 	// result in repeating a previous state that was checkpointed.
 	s.lastApplied = ""
 	s.cache.needToRetry = true
-	if status.DeepEqual(oldErrs, s.cache.errs) {
-		s.cache.reconciliationWithSameErrs++
```

Comment:
The retry attempt was based on the same set of errors, but now it is based on all types of errors. Is it an intentional change?

Reply:
Yes. It is intentional. The current retry logic is a little aggressive.